mcs/class/monodoc/Monodoc/providers/EcmaDoc.cs

```diff
@@ -31,7 +31,8 @@ internal class EcmaDoc
 		                                              IDocStorage storage,
 		                                              Dictionary<string, XElement> nsSummaries,
 		                                              Func<XElement, string> indexGenerator = null,
-		                                              IEcmaProviderFileSource fileSource = null)
+		                                              IEcmaProviderFileSource fileSource = null,
```

Comment:
Yes, as with the mutable static state, this is also another thing that I hesitantly added to avoid a big refactor. The only reason I went forward with it is because the parameters I was adding were optional, and really only useful when being called specifically from mdoc ... so it wouldn't break any other clients that might be using this API. 

But you're right that it's a code smell ... if any cause comes up again to modify this code, I will certainly refactor rather than continue adding to the parameter list.